src/EditorFeatures/Core/Implementation/IntelliSense/Completion/Controller.Session_SetModelBuilderState.cs

```diff
@@ -28,7 +28,7 @@ public void SetModelBuilderState(bool includeBuilder)
                 // already in soft select mode.
                 var softSelect = includeBuilder || model.IsSoftSelection;
 
-                if (model.SelectedItem == model.SuggestionModeItem &&
+                if (model.SelectedItemOpt == model.SuggestionModeItem &&
```

Comment:
we might want to check it SelectedItemOpt is != null.  Because effectively we're saying that if both are null, we go through this path.